src/elements/content-sidebar/activity-feed/comment-form/CommentForm.js

```diff
@@ -129,7 +132,7 @@ class CommentForm extends React.Component<Props, State> {
             <Media className={inputContainerClassNames}>
                 {!isEditing && (
                     <Media.Figure className="bcs-CommentForm-avatar">
-                        <Avatar getAvatarUrl={getAvatarUrl} user={user} />
+                        {renderAvatar(<Avatar getAvatarUrl={getAvatarUrl} user={user} />)}
```

Comment:
Another thought is, what if instead of a decorating render prop we gave the render prop full control? The `renderCommentFormAvatar` could by default render `<Avatar ... />` but if the render prop is provided we let the parent component render whatever they want

Reply:
> I hate to make it more verbose, but would it make sense to have it be renderCommentFormAvatar to be more descriptive of what it does? Initially upon reading the prop I thought it was for all avatars in the activity feed.

That's actually a great suggestion. Indeed I didn't think that there might be more avatar occurrences than just in `CommentForm`.

This only collides a bit with suggestions in [this thread](https://github.com/box/box-ui-elements/pull/2880#discussion_r859107218). Merging them both would result in `renderCommentFormAvatarWithWrapper`, which for me feels like a little too much. Can I ask you to please voice your thoughts in that other thread?

Reply:
> Another thought is, what if instead of a decorating render prop we gave the render prop full control? The renderCommentFormAvatar could by default render <Avatar ... /> but if the render prop is provided we let the parent component render whatever they want

Isn't that what happens right now? The parent component can render whatever they want. We pass `<Avatar>` to it (as `children` property), but the parent can totally ignore it and render whatever they want. Unless I'm missing something.